### PR TITLE
fix Issue 15257 - __traits(compiles, …) with malformed inline asm silently ends compilation

### PR DESCRIPTION
**Pull request by Cauterite**
https://issues.dlang.org/show_bug.cgi?id=15257

make iasm errors not end compilation
dirty hack which works for me

### CODE REVIEW

**Comment by Cauterite** [comment]
Looks like there's one particular test ( fail152.d ) which relies on the incorrect DMD output from before this patch. Not sure what to do in this situation (apart from fix the test of course).

**Comment by CyberShadow** [comment]
> apart from fix the test of course

Yes.

**Comment by Cauterite** [comment]
It might be worth merging this patch as a temporary workaround, since it seems like a clean solution would involve significant refactoring. Unless someone else thinks they're up to the task.

**Comment by WalterBright** [comment]
> dirty hack

I'd rather not. None of the rest of dmd uses exceptions, I don't think we should start now.

File: src/iasm.c [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 /*
  * Copyright (c) 1992-1999 by Symantec
- * Copyright (c) 1999-2016 by Digital Mars
+ * Copyright (c) 1999-2013 by Digital Mars
```
Comment (wilzbach) [inline src/iasm.c]: huh? back to the past? ;-)

File: src/iasm.c [hunk 2]
```diff
@@ -142,42 +144,38 @@ static const REG aregFp[] =
...
+#define _EDI            7
+#define _SI             6
+#define _ESI            6
+#define _ES             0
+#define _CS             1
+#define _SS             2
+#define _DS             3
+#define _GS             5
+#define _FS             4
+
```
Comment (wilzbach) [inline src/iasm.c]: Ehm I assume this is accidentally made it through your rebase?

Comment (Cauterite) [inline src/iasm.c]: well i haven't test it yet, but otherwise it's good to go

**Comment by Cauterite** [comment]
Will open a new pull fixing both the iasm issues i had pulls open for

**Pull request closed without merging**